storage/modules/block_storage.go

```diff
@@ -1084,6 +1184,108 @@ func (b *BlockStorage) FindTransaction(
 	return newestBlock, newestTransaction, nil
 }
 
+func (b *BlockStorage) FindRelatedTransactions(
+	ctx context.Context,
+	transactionIdentifier *types.TransactionIdentifier,
+	db database.Transaction,
+) (*types.BlockIdentifier, *types.Transaction, []*types.Transaction, error) {
+	rootBlock, tx, err := b.FindTransaction(ctx, transactionIdentifier, db)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	if rootBlock == nil {
+		return nil, nil, nil, nil
+	}
+
+	childIds, err := b.getChildren(ctx, tx, db)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// create map of seen transactions to avoid duplicates
+	seen := make(map[string]struct{})
+	children := []*types.Transaction{}
+
+	i := 0
+	for {
+		if i >= len(childIds) {
+			break
+		}
+		childID := childIds[i]
+		i++
+
+		// skip duplicates
+		if _, val := seen[childID.Hash]; !val {
```

Comment:
nit: thoughts on keeping this checking consistent with the rest of the codebase?
```suggestion
		if _, ok := seen[childID.Hash]; !ok {
```